lib/using_restricted_array.rb

```diff
@@ -6,56 +6,109 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def length(array)
-  raise NotImplementedError
+ length = 0
+ length += 1 until array[length].nil?
+ return length
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def print_array(array)
-  raise NotImplementedError
-end
+  i = 0
+  i += 1 until array[i].nil?
+ return i
+end 
```

Comment:
It looks like print_array is returning the length instead of printing the integer values?